controllers/machine_controller.go

```diff
@@ -183,9 +185,15 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	} else {
 		klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name)
-		if err := r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
-			klog.Errorf("Error deleting node %q for machine %q: %v", m.Status.NodeRef.Name, m.Name, err)
-			return ctrl.Result{}, err
+		waitErr := wait.PollImmediate(2*time.Second, 10*time.Second, func() (bool, error) {
+			if err = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return true, nil
+		})
+		if waitErr != nil {
+			r.Log.Error(err, "timed out deleting Machine's node, moving on", "node", m.Status.NodeRef.Name, "machine", m.Name)
+			r.recorder.Eventf(m, corev1.EventTypeWarning, "FailedDelete", "timed out deleting Machine's node: %v", err)
```

Comment:
```suggestion
			r.recorder.Eventf(m, corev1.EventTypeWarning, "FailedDeleteNode", "error deleting Machine's node: %v", err)
```

Reply:
Also we have pkg/util/record.Warnf. I think @vincepri wants to move to using that everywhere (right?).

Reply:
Possibly, yeah